File: backend/src/models/Event.js
class Event {
  constructor({
    id,
    title,
    description,
    dateTimeMillis,
    location,
    organizerId,
    category,
    imageUrl,
    isPastEvent,
    isCanceled,
    canceledAt,
    createdAt,
    updatedAt
  }) {
    // Modelo de evento usado antes de guardar en Firestore.
    const now = new Date().toISOString();

    this.id = id;
    this.title = title;
    this.description = description;
    this.dateTimeMillis = dateTimeMillis;
    this.location = location;
    this.organizerId = organizerId;
    this.category = category || "ACADEMIC";
    this.imageUrl = imageUrl || null;
    this.isPastEvent = Boolean(isPastEvent);
    this.isCanceled = Boolean(isCanceled);
    this.canceledAt = canceledAt || null;
    this.createdAt = createdAt || now;
    this.updatedAt = updatedAt || now;
  }
}

module.exports = Event;

File: app/src/main/java/gestion_eventos/com/domain/model/Event.kt
package gestion_eventos.com.domain.model

data class Event(
    val id: String,
    val title: String,
    val description: String,
    val dateTimeMillis: Long,
    val location: String,
    val organizerId: String,
    val category: EventCategory = EventCategory.ACADEMIC,
    val imageUrl: String? = null,
    val isPastEvent: Boolean = false,
    val isCanceled: Boolean = false,
    val canceledAt: String? = null,
    val createdAt: String? = null,
    val updatedAt: String? = null
) : Shareable {
    // Representa un evento y define como puede compartirse.
    override fun shareMessage(): String {
        return "Te invito al evento $title en $location."
    }
}
